homeassistant/components/mqtt/cover.py

```diff
@@ -178,8 +171,12 @@ def validate_options(config: ConfigType) -> ConfigType:
         vol.Optional(CONF_PAYLOAD_STOP, default=DEFAULT_PAYLOAD_STOP): vol.Any(
             cv.string, None
         ),
-        vol.Optional(CONF_POSITION_CLOSED, default=DEFAULT_POSITION_CLOSED): int,
-        vol.Optional(CONF_POSITION_OPEN, default=DEFAULT_POSITION_OPEN): int,
+        vol.Optional(CONF_POSITION_CLOSED, default=DEFAULT_POSITION_CLOSED): vol.Coerce(
+            int
+        ),
```

Comment:
Why do we need to coerce things to int?

Reply:
Because it will have floats as value when the number selector is used, and `int` will fail the validation.

Reply:
OK.. Why don't we handle that in the config flow then? `vol.Coerce` tries way too hard to convert to int in my opinion.
For example:
```py
>>> import voluptuous as vol
>>> vol.Coerce(int)("123")
123
```

Reply:
The last commit uses the validator result in case it is a str, int or float. We cannot store objects like templates.

Reply:
With the updated validator code the changes this is now resolved.